### PR TITLE
bootstrap: please document why a config change requires migration

### PR DESCRIPTION
**Pull request by RalfJung**
Currently I go to these PRs to learn how to update my config, and then I am staring at a PR description that was clearly written for the reviewer, not the random rustc developer just updating their checkout. That's no good.

### CODE REVIEW

**Comment by Kobzol** [comment]
Maybe a more "enforceable" solution would be to use a struct containing a PR number and a string with config.toml modification instructions that would be printed by bootstrap? And use this struct in the array instead of just PR numbers.

**Comment by RalfJung** [comment]
Maybe. @onur-ozkan said they are re-thinking parts of the system anyway.

I honestly don't really want to spend time of this, I just got frustrated this morning when I got two change notifications for my config.toml and none of them told me what I should do to my config.toml in order to migrate...

**Comment by RalfJung** [comment]
I also removed the duplicate printing of the changelog notice. This message is so verbose it gets in the way of regular rustc development (I can't see the output I am looking for in between all these warnings), and the message really is not important enough to justify that. (At least, the 3 times that this system was invoked so far, it really wasn't very important that everyone drop what they are doing and immediately change their config.)

File: src/bootstrap/src/lib.rs [hunk 1]
```diff
@@ -78,6 +78,9 @@ const LLD_FILE_NAMES: &[&str] = &["ld.lld", "ld64.lld", "lld-link", "wasm-ld"];
 /// If you make any major changes (such as adding new values or changing default values),
 /// please ensure that the associated PR ID is added to the end of this list.
 /// This is necessary because the list must be sorted by the merge date.
+/// Also make sure that your PR description tells people what they need to do
+/// to migrate their configuration. (If there's nothing to migrate, consider whether
+/// having *every rustc developer* open your PR page is really necessary.)
```
Comment (onur-ozkan) [inline src/bootstrap/src/lib.rs:83]: As #117815 aims to fix this problem, this change shouldn't be needed anymore.

**Comment by onur-ozkan** [comment]
> I also removed the duplicate printing of the changelog notice. This message is so verbose it gets in the way of regular rustc development (I can't see the output I am looking for in between all these warnings), and the message really is not important enough to justify that. (At least, the 3 times that this system was invoked so far, it really wasn't very important that everyone drop what they are doing and immediately change their config.)

Do we still need to do that? With #117815, this 2nd print becomes quite important. See

**Comment by RalfJung** [comment]
This entire PR is superseded by https://github.com/rust-lang/rust/pull/117815.

**Pull request closed without merging**